forgerock-auth/src/main/java/org/forgerock/android/auth/FROptions.kt

```diff
@@ -0,0 +1,234 @@
+    }
+
+    /**
+     * Build the server configurations
+     *
+     * @param block  takes the  closure to set the values for server configuration.
+     */
+    fun server(block: ServerBuilder.() -> Unit) {
+        server = ServerBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the oauth configurations
+     *
+     * @param block takes the  closure to set the values for oauth configuration.
+     */
+    fun oauth(block: OAuthBuilder.() -> Unit) {
+        oauth = OAuthBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the service configurations
+     *
+     * @param block takes the closure to set the values for service configuration.
+     */
+    fun service(block: ServiceBuilder.() -> Unit) {
+        service = ServiceBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the endpoints
+     *
+     * @param block takes the  closure to set the values for urls.
+     */
+    fun urlPath(block: UrlPathBuilder.() -> Unit) {
+        urlPath = UrlPathBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the ssl pinning
+     *
+     * @param block takes the builder closure to set the values for ssl pinning.
+     */
+    fun sslPinning(block: SSLPinningBuilder.() -> Unit) {
+        sslPinning = SSLPinningBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the custom logger
+     *
+     * @param block takes the builder closure to configure custom logger.
+     */
+    fun logger(block: LoggerBuilder.() -> Unit) {
+        logger = LoggerBuilder().apply(block).build()
+    }
+
+    private fun build(): FROptions {
+        return FROptions(server, oauth, service, urlPath, sslPinning, logger)
+    }
+
+}
+
+/**
+ * Data class for the server configurations
+ */
+data class Server(val url: String? = "",
+                  val realm: String? = "root",
+                  val timeout: Int = 30,
+                  val cookieName: String? = "iPlanetDirectoryPro")
+
+/**
+ * Server builder to build the SDK configuration information specific to server
+ */
+class ServerBuilder {
+    var url: String? = ""
+    var realm: String? = "root"
+    var timeout: Int = 30
+    var cookieName: String? = "iPlanetDirectoryPro"
+
+    fun build(): Server = Server(url, realm, timeout, cookieName)
+}
+
+/**
+ * Data class for the OAuth configurations
+ */
+data class OAuth(val oauthClientId: String? = "",
+                 val oauthRedirectUri: String? = "",
+                 val oauthScope: String? = "",
+                 val oauthThresholdSeconds: Long = 30,
+                 val oauthCacheSeconds: Long = 0,
+                 val cookieCacheSeconds: Long = 0)
+
+/**
+ * Oauth builder to build the SDK configuration information specific to oauth
+ */
+class OAuthBuilder {
+    var oauthClientId: String? = ""
+    var oauthRedirectUri: String? = ""
+    var oauthScope: String? = ""
+    var oauthThresholdSeconds: Long = 30
```

Comment:
Does the oauthThresholdSeconds need a default value?

Reply:
this was aligned with strings.xml value. the idea is we are going to replicate the values in strings.xml to FROptions for all the default values

Reply:
The Strings.xml is passing a value, if you delete that key from there that would default to nothing I guess. That's why I am asking, I think those should be optionals or 0 if possible